src/devices/vdml_adi.c

```diff
@@ -10,371 +10,110 @@
-	struct {
-		int32_t calib;
-	} analog_data;
-	struct {
-		bool was_pressed;
-	} digital_data;
-	struct {
-		bool reversed;
-	} encoder_data;
-	struct __attribute__((packed)) {
-		double multiplier;
-		double tare_value;
-	} gyro_data;
-} adi_data_s_t;
-
-#define transform_adi_port(port)       \
-	if (port >= 'a' && port <= 'h')      \
-		port -= 'a';                       \
-	else if (port >= 'A' && port <= 'H') \
-		port -= 'A';                       \
-	else                                 \
-		port--;                            \
-	if (port > 7 || port < 0) {          \
-		errno = ENXIO;                     \
-		return PROS_ERR;                   \
-	}
-
-#define validate_type(device, port, type)                                                                 \
-	adi_port_config_e_t config = (adi_port_config_e_t)vexDeviceAdiPortConfigGet(device->device_info, port); \
-	if (config != type) {                                                                                   \
-		errno = EADDRINUSE;                                                                                   \
-		return PROS_ERR;                                                                                      \
-	}
-
-#define validate_motor(device, port)                                                                      \
-	adi_port_config_e_t config = (adi_port_config_e_t)vexDeviceAdiPortConfigGet(device->device_info, port); \
-	if (config != E_ADI_LEGACY_PWM && config != E_ADI_LEGACY_SERVO) {                                       \
-		errno = EADDRINUSE;                                                                                   \
-		return PROS_ERR;                                                                                      \
-	}
-
-#define validate_twowire(port_top, port_bottom) \
-	if (abs(port_top - port_bottom) > 1) {        \
-		errno = ENXIO;                              \
-		return PROS_ERR;                            \
-	}                                             \
-	int port;                                     \
-	if (port_top < port_bottom)                   \
-		port = port_top;                            \
-	else if (port_bottom < port_top)              \
-		port = port_bottom;                         \
-	else {                                        \
-		errno = EINVAL;                             \
-		return PROS_ERR;                            \
-	}                                             \
-	if (port % 2 == 1) {                          \
-		errno = EINVAL;                             \
-		return PROS_ERR;                            \
-	}
-
-adi_port_config_e_t adi_port_get_config(uint8_t port) {
-	transform_adi_port(port);
-	claim_port_i(INTERNAL_ADI_PORT, E_DEVICE_ADI);
-	adi_port_config_e_t rtn = (adi_port_config_e_t)vexDeviceAdiPortConfigGet(device->device_info, port);
-	return_port(INTERNAL_ADI_PORT, rtn);
+adi_port_config_e_t adi_port_get_config(int32_t port) {
+	return ext_adi_port_get_config(INTERNAL_ADI_PORT, port);
```

Comment:
You expanded port to 32 bits, but you are using INTERNAL_ADI_PORT here instead of seeing if the upper bits contain that data. What are you planning on doing with this api?

Note that the upper bits being 0 will have to map to INTERNAL_ADI_PORT, instead of 22.

Reply:
This is intended to be for users of only the internal adi api, so that users will not see any change in the original api functionality.  It only takes the adi port (a-h) so there is no need to map 0 to 22.

Reply:
Then why the change moving port from 8 bits to 32?

Reply:
I'm currently working on changing smart_port and adi_port to be stored as uint8_t, so that shouldn't be an issue